File: server/internal/address/models/address.go
package models

import "server/common/middleware/database"

type Address struct {
	database.BaseEntity
	ProvinceCode string `json:"province_code"`
	CityCode     string `json:"city_code"`
	CityName     string `json:"city_name"`
	CountryCode  string `json:"country_code"`
	Keywords     string `json:"keywords"`
}

type AddressTemplate struct {
	database.BaseEntity
	UserNumId  string `json:"user_num_id"`
	AddressId  uint64 `json:"address_id"`
	TemplateId string `json:"template_id"`
}

func (a *Address) TableName() string {
	return "address"
}

func (a *AddressTemplate) TableName() string {
	return "address_template"
}

File: client/common/model/address/address.ts


export class Address {
    id: number|undefined;
    countryCode: string;
    provinceCode: string;
    cityCode: string;
    cityName: string;
    keywords: string;

    constructor(id: number|undefined, countryCode: string, provinceCode: string, cityCode: string, cityName: string, keywords: string) {
        this.id = id;
        this.countryCode = countryCode;
        this.provinceCode = provinceCode;
        this.cityCode = cityCode;
        this.cityName = cityName;
        this.keywords = keywords;
    }
}

export class AddressTemplate {
    id: number|undefined;
    resourceId: number;
    addressId: number;
    templateId: string;

    constructor(id: number|undefined, resourceId: number, addressId: number, templateId: string) {
        this.id = id;
        this.resourceId = resourceId;
        this.addressId = addressId;
        this.templateId = templateId;
    }
}